network-area-diagram/src/main/java/com/powsybl/nad/svg/SvgWriter.java

```diff
@@ -129,13 +129,57 @@ private void drawBranchEdges(Graph graph, XMLStreamWriter writer) throws XMLStre
 
             drawHalfEdge(graph, writer, edge, BranchEdge.Side.ONE);
             drawHalfEdge(graph, writer, edge, BranchEdge.Side.TWO);
+
+            if (BranchEdge.PST_EDGE.equals(edge.getType())) {
+                drawPstArrow(writer, edge);
+            }
+            if (edge.getType().equals(BranchEdge.HVDC_LINE_EDGE)) {
+                drawConverterStation(writer, edge);
+            }
+            if (BranchEdge.LINE_EDGE.equals(edge.getType()) && svgParameters.isLineNameDisplayed()) {
+                drawLineLabel(writer, edge);
+            }
+
             drawEdgeCenter(writer, edge);
 
             writer.writeEndElement();
         }
         writer.writeEndElement();
     }
 
+    private void drawLineLabel(XMLStreamWriter writer, BranchEdge edge) throws XMLStreamException {
+
+        String lineName = edge.getEquipmentId();
+
+        if (lineName != null) {
```

Comment:
You could also test if `lineName.isEmpty()`.
And to cut complexity, you'd better return if null or empty.